README.md

```diff
@@ -219,54 +367,171 @@ for (padded_string json : { string("[1, 2, 3]"), string("true"), string("[ true,
 }
 ```
 
-If you are running a server loop and want to limit the document size to keep server memory constant,
-you can set a maximum capacity:
+It's not just internal buffers though. simdjson reuses the document itself. Notice that reference?
+`document &doc`? That's key. You are only *borrowing* the document from simdjson, which purposely
+reuses and overwrites it each time you call parse. This prevent wasteful and unnecessary memory
+allocation in 99% of cases where JSON is just read, used, and converted to native values
+or thrown away.
+
+> **You are only borrowing the document from the simdjson parser. Don't keep it long term!**
+
+This is key: don't keep the `document&`, `document::element`, `document::array`, `document::object`
+or `string_view` objects you get back from the API. Convert them to C++ native values, structs and
+arrays that you own.
+
+### Keeping documents around for longer
+
+If you really need to keep parsed JSON documents around for a long time, you can **take** the
+document by declaring an actual `document` value.
 
 ```c++
-document::parser parser(1024*1024); // Set max capacity to 1MB
-for (int i=0;i<argc;i++) {
-  auto [doc, error] = parser.parse(get_corpus(argv[i]));
-  if (error == CAPACITY) { cerr << "JSON files larger than 1MB are not supported!" << endl; exit(1); }
-  if (error) { cerr << error << endl; exit(1); }
-  cout << doc;
-}
+document doc = parser.parse(json); // "steals" the document from the parser
 ```
 
-If you want absolutely constant memory usage, you can even allocate the capacity yourself at the
-beginning:
+This won't allocate anything or copy document memory: instead, it will *steal* the document memory
+from the parser. The parser will simply allocate new document memory the next time you call parse.
```

Comment:
How does the syntax "document doc = parser.parse(json);" relate to "auto [doc,error] = parser.parse(json);" which gets recommended below for preserving memory when working with large files? The doc is a reference in that case, correct? If so, it should be spelled out.